test/resolution.go

```diff
@@ -107,3 +107,57 @@ func (r *ResolvedResource) Annotations() map[string]string {
 func (r *ResolvedResource) Source() *pipelinev1beta1.ConfigSource {
 	return r.ResolvedSource
 }
+
+// RawRequest stores the raw request data
+type RawRequest struct {
+	// the request name
+	Name string
+	// the request namespace
+	Namespace string
+	// the params for the request
+	Params []pipelinev1beta1.Param
+}
+
+// Request returns a Request interface based on the RawRequest.
+func (r *RawRequest) Request() resolution.Request {
+	if r == nil {
+		r = &RawRequest{}
+	}
+	return &Request{
+		RawRequest: *r,
+	}
+}
+
+// Request implements resolution.Request and makes it easier to mock input for submit
+// Using inline structs is to avoid conflicts between field names and method names.
```

Comment:
Using inline structs is to avoid conflicts between field names and method names.